src/types.ts

```diff
@@ -125,6 +125,54 @@ export function convertPostToDTO(post: Post): tinkerforgeDTO {
   return DTO;
 }
 
+export function postKeyToTinkerforgeDTKey<K extends keyof Post>(
+  key: K,
+): keyof tinkerforgeDTO | undefined {
+  const mapping: Record<keyof Post, keyof tinkerforgeDTO | undefined> = {
+    uuid: "id",
+    label: "label",
+    date_created: "date_created",
+    date_modified: "date_modified",
+    enabled: "enabled",
+    description: "description",
+    config: "config",
+    on_connect: "on_connect",
+    uid: "uid",
+    title: undefined,
+    topic: undefined,
+    unit: undefined,
+    driver: undefined,
+    sensor_type: undefined,
+    host: undefined,
+    port: undefined,
+    pad: undefined,
+    sad: undefined,
+  };
+  return mapping[key];
+}
+
+function tinkerforgeDTKeyToPostKey<K extends keyof tinkerforgeDTO>(
+  key: K,
+): keyof Post | undefined {
+  const mapping: Record<keyof tinkerforgeDTO, keyof Post | undefined> = {
+    id: "uuid",
+    label: "label",
+    uid: "uuid",
```

Comment:
```suggestion
    id: "id",
    label: "label",
    uid: "uuid",
```
Why did you map both `id` & `uid` the with `uuid`? The DTO has an `id` key.
https://github.com/TU-Darmstadt-APQ/kraken_webui/blob/0dd8607b473c9c42b687a5c70209913acb0f7b26/src/models/zTinkerforgeSensor.schema.ts#L34-L46

Reply:
This error occurs because the Post interface keys are expected (Post has no id field): `Type '"id"' is not assignable to type 'keyof Post | undefined'.`

Reply:
so I decided to assigned uuid to both (as a placeholder until the rest of the code is rewritten)

Reply:
Yes I see the problem now. Had to look up TS records again to see it.
But nevertheless, you can map the DTOs `uid` key to the Post `uid` key.
```suggestion
    id: "uuid",
    label: "label",
    uid: "uid",
```